martian/src/lib.rs

```diff
@@ -176,33 +199,58 @@ pub fn martian_main_with_log_level<S: ::std::hash::BuildHasher>(
 
         let msg = match info.location() {
             Some(location) => format!(
-                "thread '{}' panicked at '{}': {}:{}{:?}",
-                thread,
+                "stage failed unexpectedly: '{}' {}:{}:\n{:?}",
                 msg,
                 location.file(),
                 location.line(),
                 backtrace
             ),
-            None => format!("thread '{}' panicked at '{}'{:?}", thread, msg, backtrace),
+            None => format!("stage failed unexpectedly: '{}':\n{:?}", msg, backtrace),
         };
 
+        // write to _log
         error!("{}", msg);
-        let _ = write_errors(&msg);
+
+        // write stack trace to to _stackvars. 
+        // this will just give up if any errors are encountere
+        let bt_string = format!("{:?}", backtrace);
+        let _ = File::create(&stackvars_path).map(|mut f| {
+            let _ = f.write_all(bt_string.as_bytes());
+        });
+    
+        // write to _errors
+        let _ = write_errors(&msg, false);
+
+        // call default panic handler (not sure if this is a good idea or not)
```

Comment:
As noted, there's no guarantee that this will actually get to run.

Reply:
keep file handle 3 open.